### DISCUSSION TITLE
Passive Joint setting and joint connection

### DISCUSSION THREAD

**Discussion by huiduan-25** [cyberbotics/webots#5312] [Q&A]
Hello, 

Hope everyone had a great weekend. 

I am wondering how to set a passive joint in a four bar linkage mechanism? Should I just don‘t set anything in device of HingeJoint?
Also, from the picture I attached, I am not sure how to connect the 5th joint. The shape of the selected part has been used in the 3th joint, and if I select it again, there will be two of the part? How should I solve that?

Thank you!
![image](https://user-images.githubusercontent.com/69326215/194782488-07992140-0fda-416b-9dc4-b510b92d7079.png)

**Comment by huiduan-25** [comment]
I set a SolidReference node at the endpoint of the 5th joint, it works. But 2,3,4,5 joint are fixed and does not rotate any more.
<img width="722" alt="image" src="https://user-images.githubusercontent.com/69326215/194791713-ef67f260-7c2c-49e1-b2c8-dff3660739b1.png">

**Comment by ad-daniel** [comment]
Are you sure all Solids/links present have a Physics node set?

**Reply by huiduan-25** [reply]
Hello Daniel,

I think all links have physics node set, including boundayObject and physics. I named each part and selected one for the SolidReference . I found no matter which part I select, the part will fixed. Do you know how should I solve it?

Thanks

**Reply by ad-daniel** [reply]
A similar closed-loop structure is available in the `file > open sample world > coupled_motors.wbt`, you can look at it for inspiration. Concerning your problem, it's hard to say without the world file. Can you provide your project?

**Reply by huiduan-25** [reply]
[testleg5.zip](https://github.com/cyberbotics/webots/files/9751853/testleg5.zip)
Hi daniel,

I have attached my project file.

**Reply by huiduan-25** [reply]
Hi, Daniel

I have found the solution of problem. Some of the axis of the joint were set incorrectly.

Thank you!
